### PR TITLE
YOLOv8 pipeline example

### PR DESCRIPTION
**Pull request by mgoin**
```
# Install dependencies
pip install "deepsparse[yolo]" ultralytics

# Download example code, model, and sample image
git clone https://github.com/neuralmagic/deepsparse.git
cd deepsparse/examples/ultralytics-yolov8
yolo task=detect mode=export model=yolov8n.pt format=onnx opset=13
wget -O basilica.jpg https://raw.githubusercontent.com/neuralmagic/deepsparse/main/src/deepsparse/yolo/sample_images/basilica.jpg
```

```
python annotate.py --source basilica.jpg  --model_filepath yolov8n.onnx
```

Screenshot 2023-01-11 at 6 53 46 PM

### CODE REVIEW

File: src/deepsparse/yolo/pipelines.py [hunk 1]
```diff
@@ -236,10 +236,13 @@ def process_engine_outputs(
...
 
         # NMS
-        batch_output = postprocess_nms(
-            batch_output,
-            iou_thres=kwargs.get("iou_thres", 0.25),
-            conf_thres=kwargs.get("conf_thres", 0.45),
+        from ultralytics.yolo.utils import ops
+        import torch
+
+        batch_output = ops.non_max_suppression(
```
Comment (corey-nm) [inline src/deepsparse/yolo/pipelines.py:242]: From what I can tell the format of batch_output is the same as the old nms:

```python
            (x1, y1, x2, y2, confidence, class, mask1, mask2, ...).
```
So this should be good?

https://github.com/ultralytics/ultralytics/blob/f5dfd5be8b169935266ff602b8004a34d94feec5/ultralytics/yolo/utils/ops.py#L139

Comment (mgoin) [reply src/deepsparse/yolo/pipelines.py:242]: Yep xyxy is YOLOv5 as well

**Comment by corey-nm** [comment]
Noting that I had to specify `opset=13` when running the yolo export command

**Comment by corey-nm** [comment]
Another note, exporting with `dynamic=True` sets up this input size in the onnx graph:
```
name: "images"
type {
  tensor_type {
    elem_type: 1
    shape {
      dim {
        dim_param: "batch"
      }
      dim {
        dim_value: 3
      }
      dim {
        dim_param: "height"
      }
      dim {
        dim_param: "width"
      }
    }
  }
}
```

which is making the pipeline think the image size is (0, 0) from the onnx graph

**Comment by mwitiderrick** [comment]
This code doesn't work as per my tests. 
```
2023-01-12 10:30:30 deepsparse.utils.annotate INFO     Results will be saved to annotation-results/deepsparse-annotations/deepsparse-annotations-004
DeepSparse Engine, Copyright 2021-present / Neuralmagic, Inc. version: 1.2.0 COMMUNITY EDITION | (45d54d49) (release) (optimized) (system=avx512_vnni, binary=avx512)
Traceback (most recent call last):
  File "/home/ubuntu/env/bin/deepsparse.object_detection.annotate", line 33, in <module>
    sys.exit(load_entry_point('deepsparse-nightly', 'console_scripts', 'deepsparse.object_detection.annotate')())
  File "/home/ubuntu/env/lib/python3.8/site-packages/click/core.py", line 1128, in __call__
    return self.main(*args, **kwargs)
  File "/home/ubuntu/env/lib/python3.8/site-packages/click/core.py", line 1053, in main
    rv = self.invoke(ctx)
  File "/home/ubuntu/env/lib/python3.8/site-packages/click/core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/ubuntu/env/lib/python3.8/site-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "/home/ubuntu/env/lib/python3.8/site-packages/deepsparse/yolo/annotate.py", line 212, in main
    annotated_image = annotate(
  File "/home/ubuntu/env/lib/python3.8/site-packages/deepsparse/utils/annotate.py", line 368, in annotate
    pipeline_output = pipeline(images=[image])
  File "/home/ubuntu/env/lib/python3.8/site-packages/deepsparse/pipeline.py", line 260, in __call__
    pipeline_outputs = self.process_engine_outputs(
  File "/home/ubuntu/env/lib/python3.8/site-packages/deepsparse/yolo/pipelines.py", line 256, in process_engine_outputs
    batch_class_names = [
  File "/home/ubuntu/env/lib/python3.8/site-packages/deepsparse/yolo/pipelines.py", line 257, in <listcomp>
    self.class_names[label_string]
KeyError: '8347'
```

**Comment by mgoin** [comment]
> This code doesn't work as per my tests. 
> 
> ```
> 
> 2023-01-12 10:30:30 deepsparse.utils.annotate INFO     Results will be saved to annotation-results/deepsparse-annotations/deepsparse-annotations-004
> 
> DeepSparse Engine, Copyright 2021-present / Neuralmagic, Inc. version: 1.2.0 COMMUNITY EDITION | (45d54d49) (release) (optimized) (system=avx512_vnni, binary=avx512)
> 
> Traceback (most recent call last):
> 
>   File "/home/ubuntu/env/bin/deepsparse.object_detection.annotate", line 33, in <module>
> 
>     sys.exit(load_entry_point('deepsparse-nightly', 'console_scripts', 'deepsparse.object_detection.annotate')())
> 
>   File "/home/ubuntu/env/lib/python3.8/site-packages/click/core.py", line 1128, in __call__
> 
>     return self.main(*args, **kwargs)
> 
>   File "/home/ubuntu/env/lib/python3.8/site-packages/click/core.py", line 1053, in main
> 
>     rv = self.invoke(ctx)
> 
>   File "/home/ubuntu/env/lib/python3.8/site-packages/click/core.py", line 1395, in invoke
> 
>     return ctx.invoke(self.callback, **ctx.params)
> 
>   File "/home/ubuntu/env/lib/python3.8/site-packages/click/core.py", line 754, in invoke
> 
>     return __callback(*args, **kwargs)
> 
>   File "/home/ubuntu/env/lib/python3.8/site-packages/deepsparse/yolo/annotate.py", line 212, in main
> 
>     annotated_image = annotate(
> 
>   File "/home/ubuntu/env/lib/python3.8/site-packages/deepsparse/utils/annotate.py", line 368, in annotate
> 
>     pipeline_output = pipeline(images=[image])
> 
>   File "/home/ubuntu/env/lib/python3.8/site-packages/deepsparse/pipeline.py", line 260, in __call__
> 
>     pipeline_outputs = self.process_engine_outputs(
> 
>   File "/home/ubuntu/env/lib/python3.8/site-packages/deepsparse/yolo/pipelines.py", line 256, in process_engine_outputs
> 
>     batch_class_names = [
> 
>   File "/home/ubuntu/env/lib/python3.8/site-packages/deepsparse/yolo/pipelines.py", line 257, in <listcomp>
> 
>     self.class_names[label_string]
> 
> KeyError: '8347'
> 
> ```

@mwitiderrick can you try using the latest release of DeepSparse 1.3.1?

**Comment by mgoin** [comment]
There are issues with DeepSparse 1.3.1 running YOLOv8, so we'll be publishing 1.3.2 next week with fixes

**Review by corey-nm — APPROVED** [review]
Tested with latest 1.3.2 deepsparse engine and it works. Nice example!!

**Review by bfineran — APPROVED** [review]

**Pull request merged**